Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Pipeline/Forward/ForwardPassVertexData.azsli

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// This #define magic lets you use the VsInput & VsOutput classes in this file without making it the final VsInput & VsOutput
+// used in your shader. Simply #define VsInput & VsOutput to your custom definitions before including this file
+//
+
+
+
+#ifndef VsInput
+#define VsInput         VsInput_ForwardPass
+#endif
+
+#ifndef VsSystemValues
+#define VsSystemValues  VsSystemValues_ForwardPass
+#endif
+
+#ifndef VsOutput
+#define VsOutput        VsOutput_ForwardPass
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_POSITION
+#define MATERIAL_USES_VERTEX_POSITION 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_POSITIONWS
+#define MATERIAL_USES_VERTEX_POSITIONWS 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_LOCALPOSITION
+#define MATERIAL_USES_VERTEX_LOCALPOSITION 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_PREV_POSITION
+#define MATERIAL_USES_VERTEX_PREV_POSITION 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_NORMAL
+#define MATERIAL_USES_VERTEX_NORMAL 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_TANGENT
+#define MATERIAL_USES_VERTEX_TANGENT 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_BITANGENT
+#define MATERIAL_USES_VERTEX_BITANGENT 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_UV
+#define MATERIAL_USES_VERTEX_UV 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_DETAIL_UV
+#define MATERIAL_USES_VERTEX_DETAIL_UV 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_COLOR
+#define MATERIAL_USES_VERTEX_COLOR 0
+#endif
+
+#ifndef MATERIAL_USES_VERTEX_BLENDMASK 
+#define MATERIAL_USES_VERTEX_BLENDMASK 0
+#endif
+
+#if MATERIAL_USES_VERTEX_COLOR && MATERIAL_USES_VERTEX_BLENDMASK
+#error("Only one of MATERIAL_USES_VERTEX_BLENDMASK and MATERIAL_USES_VERTEX_COLOR can be active at once")
+#endif
+
+
+// These options indicate whether the vertex input struct's "m_optional_xxx" is bound. If false, it is not safe to read from m_optional_xxx.
+// This option gets set automatically by the system at runtime; there is a soft naming convention that associates it with m_optional_xx.
+// (search "m_optional_" in ShaderVariantAssetBuilder for details on the naming convention).
+// [GFX TODO][ATOM-14475]: Come up with a more elegant way to associate the isBound flag with the input stream.
+
+#if MATERIAL_USES_VERTEX_COLOR
+    option bool o_color0_isBound = false;
+#endif
+#if MATERIAL_USES_VERTEX_BLENDMASK
+    option bool o_blendMask_isBound = false;
+#endif
+#if MATERIAL_USES_VERTEX_PREV_POSITION
+    option bool o_prevPosition_isBound = false;
+#endif
+
+
+struct VsInput_ForwardPass
+{
+    // Base fields (required by the template azsli file)...
+#if MATERIAL_USES_VERTEX_POSITION
+    float3 position : POSITION;
```

Comment:
recommend using the same naming convention as before by using "m_" for all these variables

Reply:
These structs didn't use the 'm_' naming convention before, and i didn't want to change the material-functions more than i had to, especially since several other structs in the Material-Code (Surface, PixelGeometryData, LightingData, etc) also don't follow that naming convention.